File: jdroid-java-core/src/main/java/com/jdroid/java/logger/MuteLogger.java
package com.jdroid.java.logger;

import org.slf4j.helpers.MarkerIgnoringBase;

/**
 * A logger wrapper that filter all log requests.
 */
public class MuteLogger extends MarkerIgnoringBase {

	private static final long serialVersionUID = -6526350853255198576L;

	public MuteLogger() {

	}

	@Override
	public boolean isTraceEnabled() {
		return false;
	}

	@Override
	public void trace(final String msg) {
		// Ignore
	}

	@Override
	public void trace(final String format, final Object param1) {
		// Ignore
	}

	@Override
	public void trace(final String format, final Object param1, final Object param2) {
		// Ignore
	}

	@Override
	public void trace(final String format, final Object[] argArray) {
		// Ignore
	}

	@Override
	public void trace(final String msg, final Throwable t) {
		// Ignore
	}

	@Override
	public boolean isDebugEnabled() {
		return false;
	}

	@Override
	public void debug(final String msg) {
		// Ignore
	}

	@Override
	public void debug(final String format, final Object arg1) {
		// Ignore
	}

	@Override
	public void debug(final String format, final Object param1, final Object param2) {
		// Ignore
	}

	@Override
	public void debug(final String format, final Object[] argArray) {
		// Ignore
	}

	@Override
	public void debug(final String msg, final Throwable t) {
		// Ignore
	}

	@Override
	public boolean isInfoEnabled() {
		return false;
	}

	/**
	 * @see org.slf4j.Logger#info(java.lang.String)
	 */
	@Override
	public void info(final String msg) {
		// Ignore
	}

	@Override
	public void info(final String format, final Object arg) {
		// Ignore
	}

	@Override
	public void info(final String format, final Object arg1, final Object arg2) {
		// Ignore
	}

	@Override
	public void info(final String format, final Object[] argArray) {
		// Ignore
	}

	@Override
	public void info(final String msg, final Throwable t) {
		// Ignore
	}

	@Override
	public boolean isWarnEnabled() {
		return false;
	}

	@Override
	public void warn(final String msg) {
		// Ignore
	}

	@Override
	public void warn(final String format, final Object arg) {
		// Ignore
	}

	@Override
	public void warn(final String format, final Object arg1, final Object arg2) {
		// Ignore
	}

	@Override
	public void warn(final String format, final Object[] argArray) {
		// Ignore
	}

	@Override
	public void warn(final String msg, final Throwable t) {
		// Ignore
	}

	@Override
	public boolean isErrorEnabled() {
		return false;
	}

	@Override
	public void error(final String msg) {
		// Ignore
	}

	@Override
	public void error(final String format, final Object arg) {
		// Ignore
	}

	@Override
	public void error(final String format, final Object arg1, final Object arg2) {
		// Ignore
	}

	@Override
	public void error(final String format, final Object[] argArray) {
		// Ignore
	}

	@Override
	public void error(final String msg, final Throwable t) {
		// Ignore
	}

}

File: jdroid-java-logging/src/main/java/com/dipien/java/logging/MuteLogger.kt
package com.dipien.java.logging

import org.slf4j.helpers.MarkerIgnoringBase

/**
 * A logger wrapper that filter all log requests.
 */
class MuteLogger : MarkerIgnoringBase() {

    override fun isTraceEnabled(): Boolean {
        return false
    }

    override fun trace(msg: String) {
        // Ignore
    }

    override fun trace(format: String, param1: Any) {
        // Ignore
    }

    override fun trace(format: String, param1: Any, param2: Any) {
        // Ignore
    }

    override fun trace(format: String, argArray: Array<Any>) {
        // Ignore
    }

    override fun trace(msg: String, t: Throwable) {
        // Ignore
    }

    override fun isDebugEnabled(): Boolean {
        return false
    }

    override fun debug(msg: String) {
        // Ignore
    }

    override fun debug(format: String, arg1: Any) {
        // Ignore
    }

    override fun debug(format: String, param1: Any, param2: Any) {
        // Ignore
    }

    override fun debug(format: String, argArray: Array<Any>) {
        // Ignore
    }

    override fun debug(msg: String, t: Throwable) {
        // Ignore
    }

    override fun isInfoEnabled(): Boolean {
        return false
    }

    override fun info(msg: String) {
        // Ignore
    }

    override fun info(format: String, arg: Any) {
        // Ignore
    }

    override fun info(format: String, arg1: Any, arg2: Any) {
        // Ignore
    }

    override fun info(format: String, argArray: Array<Any>) {
        // Ignore
    }

    override fun info(msg: String, t: Throwable) {
        // Ignore
    }

    override fun isWarnEnabled(): Boolean {
        return false
    }

    override fun warn(msg: String) {
        // Ignore
    }

    override fun warn(format: String, arg: Any) {
        // Ignore
    }

    override fun warn(format: String, arg1: Any, arg2: Any) {
        // Ignore
    }

    override fun warn(format: String, argArray: Array<Any>) {
        // Ignore
    }

    override fun warn(msg: String, t: Throwable) {
        // Ignore
    }

    override fun isErrorEnabled(): Boolean {
        return false
    }

    override fun error(msg: String) {
        // Ignore
    }

    override fun error(format: String, arg: Any) {
        // Ignore
    }

    override fun error(format: String, arg1: Any, arg2: Any) {
        // Ignore
    }

    override fun error(format: String, argArray: Array<Any>) {
        // Ignore
    }

    override fun error(msg: String, t: Throwable) {
        // Ignore
    }
}
